marklogic-client-api/src/main/java/com/marklogic/client/datamovement/PathSplitter.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2020 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.marklogic.client.datamovement;
+
+import com.marklogic.client.document.DocumentWriteOperation;
+import com.marklogic.client.io.marker.AbstractWriteHandle;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
```

Comment:
We should avoid incurring a compile-time dependency on org.jetbrains.annotations.*

The annotations are useful, but we want to be able to edit the sources in any IDE and also want to minimize our dependencies as much as possible.

Reply:
Got it. Removed the dependency.